src/jsonformat.erl

```diff
@@ -26,30 +26,30 @@
 
 %%%_* Types ============================================================
 -type config() :: #{ new_line => boolean()
-                   , text_output_key => atom()
+                   , key_mapping => #{atom() => atom()}
                    , format_funs => #{atom() => fun((_) -> _)}}.
 
 -export_type([config/0]).
 
 %%%_* Macros ===========================================================
 %%%_* Options ----------------------------------------------------------
 -define(NEW_LINE, false).
--define(TEXT_OUTPUT_KEY, text).
 
 %%%_* Code =============================================================
 %%%_ * API -------------------------------------------------------------
 -spec format(logger:log_event(), config()) -> unicode:chardata().
 format(#{msg:={report, #{format:=Format, args:=Args, label:={error_logger, _}}}} = Map, Config) ->
-  Report = #{text_output_key(Config) => io_lib:format(Format, Args)},
+  Report = #{text => io_lib:format(Format, Args)},
   format(Map#{msg := {report, Report}}, Config);
 format(#{level:=Level, msg:={report, Msg}, meta:=Meta}, Config) when is_map(Msg) ->
   Data0 = maps:merge(Msg, Meta#{level => Level}),
-  Data1 = format_data(Data0, Config),
-  encode(pre_encode(Data1, Config), Config);
+  Data1 = apply_key_mapping(Data0, Config),
+  Data2 = format_data(Data1, Config),
```

Comment:
Just a random though. Maybe `format_data` should be called `apply_format_funs` to  put a bit more emphasise on what the function is doing?